src/webgl/p5.RendererGL.Retained.js

```diff
@@ -98,26 +85,37 @@ p5.RendererGL.prototype.createBuffers = function(gId, obj) {
 p5.RendererGL.prototype.drawBuffers = function(gId) {
   this._setDefaultCamera();
   var gl = this.GL;
-  var shaderKey = this._getCurShaderId();
-  var shaderProgram = this.mHash[shaderKey];
-  //vertex position buffer
-  gl.bindBuffer(gl.ARRAY_BUFFER, this.gHash[gId].vertexBuffer);
-  gl.vertexAttribPointer(
-    shaderProgram.vertexPositionAttribute,
-    3, gl.FLOAT, false, 0, 0);
-  //normal buffer
-  gl.bindBuffer(gl.ARRAY_BUFFER, this.gHash[gId].normalBuffer);
-  gl.vertexAttribPointer(
-    shaderProgram.vertexNormalAttribute,
-    3, gl.FLOAT, false, 0, 0);
-  // uv buffer
-  gl.bindBuffer(gl.ARRAY_BUFFER, this.gHash[gId].uvBuffer);
-  gl.vertexAttribPointer(
-    shaderProgram.textureCoordAttribute,
-    2, gl.FLOAT, false, 0, 0);
-  //vertex index buffer
-  gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, this.gHash[gId].indexBuffer);
-  this._setMatrixUniforms(shaderKey);
+  var shaderProgram, shaderKey;
+  if(this.currentShader.vertSource === undefined ||
+     this.currentShader.fragSource === undefined) {
+    // The shader isn't loaded, so don't render anything this pass
+    return;
+  } else {
+    shaderProgram = this._setCurrentShader(this.currentShader);
+    shaderKey = this.curShaderId;
+  }
+  gl.useProgram(shaderProgram);
+
+  for(var attribName in this.gHash[gId].attributes) {
+    if(attribName !== 'index') {
+      var attribute = this.gHash[gId].attributes[attribName];
+      gl.bindBuffer(gl.ARRAY_BUFFER, attribute.buffer);
+      var attribLocation = gl.getAttribLocation(shaderProgram, attribute.name);
+      if(attribLocation !== -1) {
+        gl.enableVertexAttribArray(attribLocation);
+        gl.vertexAttribPointer(attribLocation,
+                               attribute.size, attribute.type, false, 0, 0);
+      }
+    }
+  }
+
+  //TODO: This re-binds the textures each render call, which could be more
```

Comment:
yea, let's discuss more.  I agree this seems like it could be a big bottleneck especially if we're performing video texture manipulation.